### PR TITLE
Fixes #14124: forbid empty strings in Ltac notations

### LINKED ISSUE #14124: Tactic Notation: Anomaly: Uncaught exception Failure("empty token.").
#### Description of the problem

```coq
Tactic Notation "foo" constr(y) "in" "" := idtac.
```

`Error: Anomaly "Uncaught exception Failure("empty token.")."`
`Please report at http://coq.inria.fr/bugs/.`

#### Coq Version

`The Coq Proof Assistant, version 8.13.1 (March 2021)
compiled on Mar 4 2021 14:01:37 with OCaml 4.11.1`

This probably used to work in 8.4 or so https://github.com/coq/coq/issues/4494#issuecomment-337538890

### PR DESCRIPTION
**Pull request by herbelin**
**Kind:** bug fix

Fixes / closes #14124

- [x] Added / updated test-suite
- [x] Entry added in the changelog

### CODE REVIEW

File: plugins/ltac/g_ltac.mlg [hunk 1]
```diff
@@ -450,10 +450,13 @@ let pr_ltac_production_item = function
   in
   str arg ++ str "(" ++ Id.print id ++ sep ++ str ")"
 
+let check_non_empty_string ?loc s =
+  if s = "" then CErrors.user_err ?loc (str "Invalid empty string.")
```
Comment (ppedrot) [inline plugins/ltac/g_ltac.mlg:454]: Nit: use `String.is_empty` instead.

**Review by ppedrot — APPROVED** [review]

**Comment by ppedrot** [comment]
Adding a test is probably a good idea before I merge though.

**Comment by herbelin** [comment]
> Adding a test is probably a good idea before I merge though.

The problem is that I don't think we have a way to test parsing errors (they are not caught by `Fail`).

**Comment by ppedrot** [comment]
Oh, right. You could always write a misc test but it could legitimately be considered overblown. Let's merge as-is then.

**Pull request merged**